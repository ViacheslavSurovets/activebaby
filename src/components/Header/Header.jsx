import React from 'react';
import { withRouter } from 'react-router-dom';

import {
  HeaderMainContainer,
  HeaderInfoContainer,
  HeaderItemsWrapperLeft,
  HeaderItemsWrapperRight,
  HeaderItem,
  ShopCartWrapper,
  LogoTabletWrapper,
  StyledLink
} from './styles';
import {
  LogoTablet,
  SearchMobile,
  ShopCart,
  LoginHeaderLink,
  HeaderNavContainer,
  LocationMarkerLink,
  BurgerMenu
} from '@components';
import PropTypes from 'prop-types';


const Header = ( { history } ) => {
  return (
    <HeaderMainContainer>
      <HeaderInfoContainer>
        <HeaderItemsWrapperLeft>
          { /*desktop*/ }
          <HeaderItem>
            <StyledLink href='tel: +375 33 3765823'>+375 33 3765823</StyledLink>
          </HeaderItem>
          <HeaderItem>
            <StyledLink href='mailto: strollers@gmail.com'>E-MAIL</StyledLink>
          </HeaderItem>
          <HeaderItem>
            <LocationMarkerLink />
          </HeaderItem>
          { /*Laptop and mobile*/ }

          <BurgerMenu />

        </HeaderItemsWrapperLeft>

        { /*Laptop / mobile*/ }
        <LogoTabletWrapper onClick={ () => history.push ( '/' ) }>
          <LogoTablet />
        </LogoTabletWrapper>

        <HeaderItemsWrapperRight>
          { /*desktop*/ }
          <LoginHeaderLink />
          { /* tablet / mobile*/ }
          <SearchMobile />
          { /*desktop / laptop / mobile*/ }
          <ShopCartWrapper>
            <ShopCart />
          </ShopCartWrapper>

        </HeaderItemsWrapperRight>
      </HeaderInfoContainer>

      <HeaderNavContainer />

    </HeaderMainContainer>
  );
};


export default withRouter ( Header );

Header.propTypes = {
  history: PropTypes.object
};
