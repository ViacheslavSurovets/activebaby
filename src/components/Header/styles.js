import styled, { css } from 'styled-components';
import { desktop, mobile, tablet } from '@utils/media';
import { NavLink } from 'react-router-dom';
import { searchMediaQueryLow } from '../SearchDesktop/Search';

export const styleLink = css`
  display: flex;
  align-items: center;
  user-select: none;
  text-decoration: none;
  color: ${ ( { theme } ) => theme.header.linkColor };
  transition: all .3s ease;
  font-size: 1.8rem;
    &:hover{
      color: ${ ( { theme } ) => theme.header.colorOnHover };
      transition: all .3s ease;
    }
`;

export const HeaderMainContainer = styled.header`
  ${tablet},
  ${mobile}{
  position: sticky;
    top: 0;
    z-index: 10;
  }
`;

//-------------------------------- Header Info Container Black

const styleHeaderInfoContainer = css`
  display: flex;
  flex:1;
  justify-content: space-between;
  background-color: ${ ( { theme } ) => theme.header.infoContainerBackground }; 
   font-size: 1.8rem;
`;

export const HeaderInfoContainer = styled.div`
  ${ styleHeaderInfoContainer };
    height: 3.5rem;
  `;
// Header Item Wrappers
const styleHeaderItemsWrapper = css`
  color: ${ ( { theme } ) => theme.white };
  display: flex;
  width: 40%;
  height: 100%;
  flex-direction: row;
  align-items: center;
`;

export const HeaderItemsWrapperRight = styled.div`
  ${ styleHeaderItemsWrapper };
  justify-content: flex-end;
`;

export const HeaderItemsWrapperLeft = styled.div`
  ${ styleHeaderItemsWrapper };
  justify-content: space-around;
`;

// Header Items
export const styleHeaderItem = css`
  display: flex;
  align-items: center;
  height: 100%;
  cursor: pointer;
  user-select: none;
    &:hover{
      color:  ${ ( { theme } ) => theme.header.colorOnHover };
    }
`;

export const ShopCartWrapper = styled.div`
${ styleHeaderItem };
 height: 100%;
 width: 4rem;
  ${ desktop }{
      width: 6rem;      
  }
`;

export const HeaderItem = styled.div`
  ${ mobile }{display: none}
  ${ tablet }{display: none}
  ${ desktop }{
    ${ styleHeaderItem };
  }
`;

//-------------------------------- Header Nav Container White

export const HeaderNavWrapper = styled.div`

`;

export const HeaderNavLink = styled ( NavLink )`
  ${ styleLink };
  border-bottom: 1px solid ${ ( { theme } ) => theme.white };
  height: 6rem;
  &:hover{
    border-bottom: 1px solid ${ ( { theme } ) => theme.header.colorOnHover };
  }
`;

export const DropdownLink = styled ( NavLink )`
  ${ styleLink };
  height: 5rem;
`;

export const HeaderNav = styled.nav`
  display: none;
  width: 100%;
  background-color: ${ ( { theme } ) => theme.header.headerNavBackground };
  position: relative;
  ${ desktop }{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}
`;

export const List = styled.ul`
  display: flex;
  flex: 1;
  justify-content: space-around;
  flex-wrap: wrap;
  list-style: none;
`;

export const Dropdown = styled.ul`
  opacity: 0;
  visibility: hidden;
  transition-property: opacity, visibility; 
  transition-duration: 0.4s; 
  transition-delay: 0.15s; 
  border-top: 1rem solid ${ ( { theme } ) => theme.white };
  position: absolute;
  text-align: justify;
  list-style: none;
  background: ${ ( { theme } ) => theme.header.dropdownBackground };
  width: 100%;
  min-width: 50%;
  left: 0;
  z-index:1;
`;

export const DropdownGroupWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
`;

export const DropdownGroup = styled.div`
li:first-child{
  margin-bottom: 1rem;
  border-bottom: 1px solid ${ ( { theme } ) => theme.header.borderBottomDropdown };
}
li:first-child:hover{
  border-bottom: 1px solid ${ ( { theme } ) => theme.header.borderBottomDropdownHover };
}
`;

export const DropdownGroupArticle = styled.div`
`;


export const ListItem = styled.li`
  z-index: 2;
      &:hover{
    ul{
      opacity: 1; 
      visibility: visible;
    }
  }
`;

export const SearchItemDesktop = styled.div`
  display: flex;
  flex-basis: 10rem;
  justify-content: center;
  
  &:hover {
    ${ searchMediaQueryLow }{
      div div:nth-child(2){
      display: flex;
      }
    }
  }
`;

//---------------------------------------Logo

export const LogoDesktopWrapper = styled.div`
  ${ desktop }{
    display: flex;
    height: 8rem;
  }
`;

export const LogoTabletWrapper = styled.div`
  display: flex;
  align-items: center;
`;
