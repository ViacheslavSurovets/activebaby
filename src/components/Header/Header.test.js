import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';


describe ( 'Header', () => {
  it ( 'Header Snapshot', () => {
    expect ( shallow ( <Header /> ) ).toMatchSnapshot ();
  } );
} );
