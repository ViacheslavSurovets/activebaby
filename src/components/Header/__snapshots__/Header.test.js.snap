// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Snapshot 1`] = `ShallowWrapper {}`;
